src/index.js

```diff
@@ -21,9 +21,35 @@ if (!isProduction) {
 function run() {
   const promiseResolver = isProduction ? resolveBlogPostsDataS3 : resolveBlogPostsDataLocal;
 
-  getBlogPostsData()
-    .then(promiseResolver)
-    .catch(handleError);
+  Promise.all([getMediaData(), getAuthorData()]).then(results => {
+    let mediaPromisesArray = results[0];
+    let authorPromisesArray = results[1];
+    let blogPostsData = getBlogPostsData();
+
+    // Resolve array of promises for media data, and map to each blog post.
+    Promise.all(mediaPromisesArray).then(mediaResults => {
+      blogPostsData.then(blogPosts => {
+        blogPosts.map(blogPost => {
+          blogPost.media_details = mediaResults.find((mediaItem) => { // eslint-disable-line camelcase
+            return mediaItem.id === blogPost.id;
+          }).media_details; // eslint-disable-line camelcase
+        });
+      });
+    });
+
+    // Resolve array of promises for author data, and map to each blog post.
+    Promise.all(authorPromisesArray).then(authorResults => {
+      blogPostsData.then(blogPosts => {
+        blogPosts.map(blogPost => {
+          blogPost.author_name = authorResults.find((authorItem) => { // eslint-disable-line camelcase
+            return authorItem.id === blogPost.id;
+          }).author;
+        });
+
+        promiseResolver(blogPosts);
```

Comment:
since we're using `Promise.all` for both media and author, can the `promiseResolver` (not sure if that's even the right name anymore) come at the end?  e.g.

```javascript
Promise.all([getMediaData(), getAuthorData()]).then(results => {
 let mediaPromisesArray = results[0];
 let authorPromisesArray = results[1];
 let blogPostsData = getBlogPostsData();
 
 Promise.all(mediaPromisesArray).then(mediaResults => {
   // code
 });

  Promise.all(authorPromisesArray).then(authorResults => {
    // code
  })

  promiseResolver(blogPosts);
```
?

Reply:
I couldn't find a way to put it at the end, since blogPostsData is also a promise. I feel like there must be a better way that is more DRY, but I couldn't find any better solutions.

Reply:
Yeah, I have no concrete way to do it better atm, but I tested and the data is looking good!